### ISSUE TITLE
Continuous request sizing on v1.101 and v1.102 always fails to unmarshal recommendations

### ISSUE THREAD

**Issue by rb-matt-barker** [kubecost/cost-analyzer-helm-chart#2136]
# For reproducible bugs, please provide the following

**Describe the bug**  
In versions of Kubecost >=v1.101 we're seeing a problem where we hit a run loop error while querying using annotations, reverting to 100.2 fixes the issue we're seeing

**To Reproduce**  
1. Using version 1.102.0
2. Set annotations to:
```
request.autoscaling.kubecost.com/enabled: true
request.autoscaling.kubecost.com/recommendationQueryWindow: 2d
```
3. See error (in screenshot)

**Expected behavior**  
Autoscaler annotations apply correctly and work fine

**Screenshots**  
![image](https://user-images.githubusercontent.com/128372666/231176570-6941d3bd-b48f-45ef-85b0-43d567207129.png)

**Collect logs (please complete the following information):**
```
2023-04-06T11:27:49Z ERR Run loop attempt failed consecutively. Waiting a bit longer before retrying. error="failed to create sizing patch for workload 'monitoring/deployment/pubsubmonitor': failed to get recommendation for '{Namespace:monitoring ControllerKind:deployment ControllerName:pubsubmonitor ContainerName:pubsubmonitor}' '{Window:2d TargetUtilizationCPU:<nil> TargetUtilizationMemory:<nil>}': unmarshaling recommendation: json: cannot unmarshal object into Go value of type []recommendationservice.Recommendation"
```

**Please share the support case, if any**  
This is most commonly a Zendesk ticket number or link.
Ticket #3806

**What impact will this have on your ability to get value out of Kubecost?**
Can't use autoscaling feature

**Comment by AjayTripathy** [comment]
@michaelmdresser can you help triage?

**Comment by michaelmdresser** [comment]
I'm almost certain this is occurring because the API return type of `/savings/requestSizingV2` changed in https://github.com/kubecost/kubecost-cost-model/pull/1196 but cluster-controller's return type (https://github.com/kubecost/cluster-controller/blob/4517a3c5ce212fad568a98fa6683d25854d29ae8/pkg/kubescaler/resourcescaling/recommendationservice/types.go#L10) was not updated.

Impact: Continuous request sizing is probably broken for all users of v1.101 and v1.102 of Kubecost.

The fix is to update the type in cluster-controller to match the change in KCM.

**Comment by michaelmdresser** [comment]
For that matter, I suspect 1-click request sizing is _also_ broken for the same reason, see its type definitions for the API here: https://github.com/kubecost/cluster-controller/blob/4517a3c5ce212fad568a98fa6683d25854d29ae8/pkg/requestsizer/types.go#L34-L43

**Comment by AjayTripathy** [comment]
Ok, this should be a quick fix that can run out of band with the main kubecost pod release. Can we cut a fix of cluster-controller soon?

**Comment by michaelmdresser** [comment]
I don't have bandwidth to address this without dropping something else. Perhaps @teevans can help get this assigned.

**Comment by teevans** [comment]
I can likely get this assigned this week, let me find someone who can jump in .

**Comment by nik-kc** [comment]
I've got a PR in for this [here](https://github.com/kubecost/cluster-controller/pull/36).

**Comment by michaelmdresser** [comment]
Hi @rb-matt-barker this will be fixed by default in the next major release of Kubecost (v1.103.0) but you can get the fix in v1.101 and v1.102 of Kubecost by manually setting the Cluster Controller image to an updated, fixed version via the Helm value: `--set clusterController.image=gcr.io/kubecost1/cluster-controller:v0.7.0`
